meta-phosphor/conf/machine/include/obmc-bsp-common.inc

```diff
@@ -13,7 +13,7 @@ UBOOT_SUFFIX ?= "bin"
 
 MACHINEOVERRIDES =. "openbmc:"
 
-IMAGE_FSTYPES += "overlay"
+MACHINE_FEATURES += "image-overlay"
```

Comment:
> > I don't know what the best approach is here. 'overlay' and 'openpower-pflash' do not seem like distro features. 'overlay' could be considered an image feature,
> 
> For clarity its not an image feature either.  Again, features of any kind determine what packages go in the rootfs.

Is this entirely true?  Don't features also change the behavior of
packages?

In curl%.bb:
```
PACKAGECONFIG ??= "${@bb.utils.contains("DISTRO_FEATURES", "ipv6",
"ipv6", "", d)} gnutls proxy zlib"
```

So this is using a feature (ipv6) to add additional support within a
package and not adding packages themselves.

> > but it also needs some component of the machine with it at this point in time because qemu doesn't support the overlay partitioning. Looking for input.
> 
> When you say qemu do you mean the qemuarm machine or qemu in general?  There shouldn’t be a problem with qemuarm because it doesn’t pick up openpower.inc.  The problem is the core-image recipes on real hw right?

I meant that qemuarm/qemux86 with a obmc-phosphor-image do not(?) support
overlay.  Maybe this is not true?  Before I started it was not set up
that way.

> 
> What we are doing here with the overlay is exactly what IMAGE_FSTYPES is there for.  I think the most correct thing to do would be to make the image-overlay class itself do the right thing when building a core image, whether that be degrading somehow, pointing at the right sub-images, or just not doing anything,  etc.
> 

Before I started there were two issues:
1. obmc-machine with images other than obmc-phosphor-image were still setting
IMAGE_FSTYPES += "overlay" (in openpower.inc)  but there were not tasks to
fulfill this.  bitbake in 2.1 errored out even when we weren't even
trying to build those images.

2. qemuarm / qemux86 do not currently build IMAGE_FSTYPES += "overlay"
because they don't get it from openpower.inc.  Also, these targets do
not have the cramfs ro-fs, so it probably doesn't make sense to even
attempt to build an overlay.

I tried to solve #1 without changing #2.  That is the change you see
here:
https://github.com/openbmc/openbmc/pull/397/commits/511556e205480ebf3cf7ab986ef1f5b1e57da9a0

I see two (relatively easy) options:
a. Change MACHINE_FEATURES += "image-overlay" to something like
OBMC_PHOSPHOR_IMAGE_OVERLAY=1 or OBMC_PHOSPHOR_IMAGE_FEATURES +=
"image-overlay".  -- Same effective behavior, different names.

b. Change obmc-phosphor-image.bbclass to always IMAGE_FSTYPES +=
"overlay" directly in there. -- Changes qemuarm / qemux86 behavior.

-- 
Patrick Williams